README.md

```diff
@@ -82,3 +83,26 @@ source/bolts/experimental/signatures.d(310,5): Error: static assert:  "Type MyRa
   source/bolts/experimental/signatures.d(464): <-- Signature InputRange!(int) defined here.
   source/bolts/experimental/signatures.d(471): <-- Checked here."
 ```
+
+## Refraction (experimental):
+
+It is sometimes necessary to create a function which is an exact copy of
+another function. Or sometimes it is necessary to introduce a few variations,
+while carrying all the other aspects. Because of function attributes, parameter
+storage classes and user-defined attributes, this requires building a string
+mixin. In addition, the mixed-in code must refer only to local names, if it is
+to work across module boundaires. This module facilitates the creation of such
+mixins.
+
+For example, this creates a function that has a different name and return type,
+but retains the 'pure' attribute from the original function:
+
+```d
+pure int answer() { return 42; }
+mixin(
+  refract!(answer, "answer").setName("realAnswer")
```

Comment:
`withName`, `withReturnType`, etc.

Reply:
> `withName`, `withReturnType`, etc.

Aaargh. Of course READMEs never fail to compile ;-)